src/ClassicUO.IO/DataReader.cs

```diff
@@ -34,76 +34,25 @@
 using System;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
 
 namespace ClassicUO.IO
 {
     /// <summary>
     ///     A fast Little Endian data reader.
     /// </summary>
-    public unsafe class DataReader : IDisposable
+    public sealed unsafe class DataReader
```

Comment:
If we aren't storing this on the heap anymore, we can change it to a `ref struct` and use a Span<byte>.. Then create a constructor and pass PinnedBuffer. This would keep the reader as generic/reusable as possible.

Usually for convenience, classes like PinnedBuffer expose some combination of `AsSpan()`, `AsSpan(count)`, `AsSpan(start, count)` methods for exactly this purpose.

Reply:
Store what on the heap? Nothing changes in this PR regarding where data is allocated.

Reply:
Yes, I was thinking the same, but that would be a possible next step.

Reply:
Ok no problem, we can do that in the next iteration.